src/language_features/semantic_highlighting.rs

```diff
@@ -19,27 +20,16 @@ pub fn semantic_highlighting_notification(params: Params, ctx: &mut Context) {
         Some(meta) => meta,
         None => return,
     };
-    let scopes = ctx
-        .capabilities
-        .as_ref()
-        .unwrap()
-        .semantic_highlighting
-        .as_ref()
-        .unwrap()
-        .scopes
-        .as_ref()
-        .expect("Server sent semantic highlight notification without setting capability");
+    let faces = &ctx.semantic_highlighting_faces;
     let ranges = params
         .lines
         .iter()
         .flat_map(|info| {
             let line: u64 = info.line as u64;
             let offset_encoding = &ctx.offset_encoding;
             info.tokens.iter().map(move |t| {
-                let scope = scopes
-                    .get(t.scope as usize)
+                let face = faces.get(t.scope as usize)
                     .expect("Semantic highlighting token sent for out-of-range scope");
```

Comment:
Does it mean kak-lsp would crash if the scope is not mapped to face in the config? Would it be better to log an error message (probably even into Kakoune's `*debug*` buffer) and skip this token?

Reply:
Ah, you are mapping scopes advertised by the language server but not configured to empty strings in `make_scope_map`. So case when face is not found *is* exceptional.